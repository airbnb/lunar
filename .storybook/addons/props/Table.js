import { styled } from '@storybook/theming';

export default styled.table(({ theme }) => ({
  width: '100%',
  maxWidth: '100%',
  margin: 0,
  padding: 0,
  border: `1px solid ${theme.appBorderColor}`,
  borderCollapse: 'collapse',
  borderSpacing: 0,
  borderRadius: theme.appBorderRadius,

  '& th': {
    textAlign: 'left',
    backgroundColor: theme.barBg,
  },

  '& td, & th': {
    border: `1px solid ${theme.appBorderColor}`,
    padding: 8,
  },
}));
