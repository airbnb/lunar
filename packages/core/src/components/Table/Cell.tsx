import React from 'react';
import withStyles, { css, WithStylesProps } from '../../composers/withStyles';

export type Props = {
  /** Display in the center horizontally. */
  centerAlign?: boolean;
  /** Span multiple columns. */
  colSpan?: number;
  /** Display horizontally at the end. */
  endAlign?: boolean;
  /** Render as a table header. */
  header?: boolean;
  /** Display horizontally at the start. */
  startAlign?: boolean;
  /** Truncate text and display all on hover. */
  truncate?: boolean;
  /** Wrap text and white space. */
  wrap?: boolean;
};

/** An individual table cell. */
export class TableCell extends React.Component<Props & WithStylesProps> {
  static defaultProps = {
    centerAlign: false,
    endAlign: false,
    header: false,
    startAlign: false,
    truncate: false,
    wrap: false,
  };

  render() {
    const {
      centerAlign,
      children,
      header,
      startAlign,
      endAlign,
      styles,
      truncate,
      wrap,
      ...props
    } = this.props;
    const Tag = header ? 'th' : 'td';

    return (
      <Tag
        {...props}
        {...css(
          truncate && styles.cell_truncate,
          startAlign && styles.cell_left,
          centerAlign && styles.cell_center,
          endAlign && styles.cell_right,
          wrap && styles.cell_wrap,
        )}
      >
        {children}
      </Tag>
    );
  }
}

export default withStyles(() => ({
  cell_truncate: {
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',

    ':hover': {
      maxWidth: 'none',
      overflow: 'inherit',
      whiteSpace: 'inherit',
      wordWrap: 'break-word',
    },
  },

  cell_left: {
    textAlign: 'left',
  },

  cell_center: {
    textAlign: 'center',
  },

  cell_right: {
    textAlign: 'right',
  },

  cell_wrap: {
    whiteSpace: 'normal',
    wordWrap: 'break-word',
  },
}))(TableCell);
