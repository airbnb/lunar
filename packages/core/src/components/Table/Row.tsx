import React from 'react';
import { mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
import withStyles, { WithStylesProps } from '../../composers/withStyles';
import { STATUSES } from '../../constants';

const statusPropType = mutuallyExclusiveTrueProps(...STATUSES);

export type Props = {
  /** Table cells to render. */
  children: NonNullable<React.ReactNode>;
  /** Dangerous/failure row (red). */
  danger?: boolean;
  /** Informational row (blue). */
  info?: boolean;
  /** Muted/disabled row (gray). */
  muted?: boolean;
  /** Notice row. */
  notice?: boolean;
  /** Successful row (green). */
  success?: boolean;
  /** Warning row (yellow). */
  warning?: boolean;
};

/** A table row that contains multiple cells. */
export class TableRow extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    danger: statusPropType,
    info: statusPropType,
    muted: statusPropType,
    notice: statusPropType,
    success: statusPropType,
    warning: statusPropType,
  };

  static defaultProps = {
    danger: false,
    info: false,
    muted: false,
    notice: false,
    success: false,
    warning: false,
  };

  render() {
    const {
      cx,
      children,
      danger,
      info,
      muted,
      notice,
      styles,
      success,
      warning,
      ...props
    } = this.props;

    return (
      <tr
        {...props}
        className={cx(
          danger && styles.row_danger,
          info && styles.row_info,
          muted && styles.row_muted,
          notice && styles.row_notice,
          success && styles.row_success,
          warning && styles.row_warning,
        )}
      >
        {children}
      </tr>
    );
  }
}

export default withStyles(({ color }) => {
  function createRow(hex: string) {
    return {
      // Overrides table specificity
      '@selectors': {
        ':nth-child(n) > td': {
          backgroundColor: hex,
        },
      },
    };
  }

  return {
    row_danger: createRow(color.core.danger[0]),
    row_info: createRow(color.core.primary[0]),
    row_muted: createRow(color.core.neutral[0]),
    row_notice: createRow(color.core.secondary[0]),
    row_success: createRow(color.core.success[0]),
    row_warning: createRow(color.core.warning[0]),
  };
})(TableRow);
