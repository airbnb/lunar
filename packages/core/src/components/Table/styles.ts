import { StyleBlock } from 'aesthetic';
import { StyleSheet } from '../../hooks/useStyles';

function createCell(styles: StyleBlock) {
  return {
    '@selectors': {
      ':only-child td': styles,
      ':only-child th': styles,
    },
  };
}

export const styleSheet: StyleSheet = ({ color, ui, unit }) => ({
  table: {
    width: '100%',
    maxWidth: '100%',
    margin: 0,
    padding: 0,
    backgroundColor: color.accent.bg,
    border: '1px solid transparent',
    borderCollapse: 'collapse',
    borderSpacing: 0,

    '@selectors': {
      ':only-child td': {
        padding: unit * 1.5,
        verticalAlign: 'top',
      },

      ':only-child th': {
        padding: unit * 1.5,
        verticalAlign: 'bottom',
        whiteSpace: 'nowrap',
      },
    },
  },

  table_bordered: {
    border: ui.border,
  },

  table_compact: {
    ...createCell({
      padding: unit,
    }),
  },

  table_horizontal: {
    '@selectors': {
      ':only-child > tbody > tr > td': {
        borderTop: ui.border,
      },
    },
  },

  table_fixed: {
    display: 'table',
    tableLayout: 'fixed',
  },

  table_loading: {
    pointerEvents: 'none',
    opacity: 0.5,
  },

  table_striped: {
    '@selectors': {
      ':only-child > tbody > tr': {
        backgroundColor: color.accent.bg,

        '@selectors': {
          ':nth-child(odd)': {
            backgroundColor: color.accent.bgHover,
          },
        },
      },
    },
  },

  table_transparent: {
    backgroundColor: 'transparent',
  },

  table_vertical: {
    ...createCell({
      borderLeft: ui.border,
      borderRight: ui.border,
    }),
  },

  responsive_wrapper: {
    maxWidth: '100%',
    overflowX: 'auto',
  },
  content_middle_align: {
    '@selectors': {
      ':only-child td': {
        verticalAlign: 'middle',
      },
    },
  },
});

export const styleSheetCell: StyleSheet = () => ({
  cell_truncate: {
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',

    ':hover': {
      maxWidth: 'none',
      overflow: 'inherit',
      whiteSpace: 'inherit',
      wordWrap: 'break-word',
    },
  },

  cell_left: {
    textAlign: 'left',
  },

  cell_center: {
    textAlign: 'center',
  },

  cell_right: {
    textAlign: 'right',
  },

  cell_wrap: {
    whiteSpace: 'normal',
    wordWrap: 'break-word',
  },
});

export const styleSheetRow: StyleSheet = ({ color }) => {
  function createRow(hex: string) {
    return {
      // Overrides table specificity
      '@selectors': {
        ':nth-child(n) > td': {
          backgroundColor: hex,
        },
      },
    };
  }

  return {
    row_danger: createRow(color.core.danger[0]),
    row_info: createRow(color.core.primary[0]),
    row_muted: createRow(color.core.neutral[0]),
    row_notice: createRow(color.core.secondary[0]),
    row_success: createRow(color.core.success[0]),
    row_warning: createRow(color.core.warning[0]),
  };
};
