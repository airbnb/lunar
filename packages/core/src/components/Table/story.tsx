import React from 'react';
import Text from '../Text';
import MenuToggle from '../MenuToggle';
import { Item as MenuItem } from '../Menu';
import Table, { Cell, Row } from '.';

export default {
  title: 'Core/Table',
  parameters: {
    inspectComponents: [Table, Cell, Row],
  },
};

export function displayATableWithNoStyles() {
  return (
    <Text>
      <Table>
        <thead>
          <tr>
            <th>One</th>
            <th>Two</th>
            <th>Three</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Title 1</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 2</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 3</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
        </tbody>
      </Table>
    </Text>
  );
}

displayATableWithNoStyles.story = {
  name: 'Display a table with no styles.',
};

export function withABorderAndVerticalDividers() {
  return (
    <Text>
      <Table bordered vertical>
        <thead>
          <tr>
            <th>One</th>
            <th>Two</th>
            <th>Three</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Title 1</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 2</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 3</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
        </tbody>
      </Table>
    </Text>
  );
}

withABorderAndVerticalDividers.story = {
  name: 'With a border and vertical dividers.',
};

export function withABorderAndHorizontalDividers() {
  return (
    <Text>
      <Table bordered horizontal>
        <thead>
          <tr>
            <th>One</th>
            <th>Two</th>
            <th>Three</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Title 1</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 2</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 3</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
        </tbody>
      </Table>
    </Text>
  );
}

withABorderAndHorizontalDividers.story = {
  name: 'With a border and horizontal dividers.',
};

export function withStripedRowsAndHorizontalDividers() {
  return (
    <Text>
      <Table horizontal striped>
        <thead>
          <tr>
            <th>One</th>
            <th>Two</th>
            <th>Three</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Title 1</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 2</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 3</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
        </tbody>
      </Table>
    </Text>
  );
}

withStripedRowsAndHorizontalDividers.story = {
  name: 'With striped rows and horizontal dividers.',
};

export function aCompactViewWithMultipleStylesApplied() {
  return (
    <Text>
      <Table compact bordered horizontal vertical striped>
        <thead>
          <tr>
            <th>One</th>
            <th>Two</th>
            <th>Three</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Title 1</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 2</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
          <tr>
            <td>Title 3</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
        </tbody>
      </Table>
    </Text>
  );
}

aCompactViewWithMultipleStylesApplied.story = {
  name: 'A compact view with multiple styles applied.',
};

export function withDifferentRowStatusColors() {
  return (
    <Text>
      <Table bordered horizontal vertical>
        <thead>
          <tr>
            <th>One</th>
            <th>Two</th>
            <th>Three</th>
          </tr>
        </thead>
        <tbody>
          <Row muted>
            <td>Title 1</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </Row>
          <Row danger>
            <td>Title 2</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </Row>
          <Row info>
            <td>Title 3</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </Row>
          <Row notice>
            <td>Title 4</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </Row>
          <Row success>
            <td>Title 5</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </Row>
          <Row warning>
            <td>Title 6</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </Row>
        </tbody>
      </Table>
    </Text>
  );
}

withDifferentRowStatusColors.story = {
  name: 'With different row status colors.',
};

export function cellsWithDifferentAlignments() {
  return (
    <Text>
      <Table bordered horizontal vertical>
        <thead>
          <tr>
            <Cell header startAlign>
              One
            </Cell>
            <Cell header centerAlign>
              Two
            </Cell>
            <Cell header endAlign>
              Three
            </Cell>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Title 1</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>Consectetur adipiscing elit.</td>
          </tr>
        </tbody>
      </Table>
    </Text>
  );
}

cellsWithDifferentAlignments.story = {
  name: 'Cells with different alignments.',
};

export function cellsWithCenteredContent() {
  return (
    <Text>
      <Table middleAlign>
        <thead>
          <tr>
            <th>One</th>
            <th>Two</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Title 1</td>
            <td>
              <div>Lorem ipsum dolor sit amet. </div>
              <div>Consectetur adipiscing elit. </div>
              <div>Lorem ipsum dolor sit amet. </div>
              <div>Consectetur adipiscing elit. </div>
              <div>Lorem ipsum dolor sit amet. </div>
              <div>Consectetur adipiscing elit. </div>
            </td>
          </tr>
          <tr>
            <td>Title 2</td>
            <td>
              <div>Lorem ipsum dolor sit amet. </div>
              <div>Consectetur adipiscing elit. </div>
              <div>Lorem ipsum dolor sit amet. </div>
              <div>Consectetur adipiscing elit. </div>
              <div>Lorem ipsum dolor sit amet. </div>
              <div>Consectetur adipiscing elit. </div>
            </td>
          </tr>
        </tbody>
      </Table>
    </Text>
  );
}

cellsWithCenteredContent.story = {
  name: 'Cells with centered content.',
};

export function withoutResponsiveWrapper() {
  return (
    <Text>
      <Table noWrap>
        <thead>
          <tr>
            <th>One</th>
            <th>Two</th>
            <th>Three</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Title 1</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>
              <MenuToggle closeOnClick accessibilityLabel="Menu" toggleLabel="Actions">
                <MenuItem>
                  <Text>Item</Text>
                </MenuItem>
              </MenuToggle>
            </td>
          </tr>
          <tr>
            <td>Title 2</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>
              <MenuToggle closeOnClick accessibilityLabel="Menu" toggleLabel="Actions">
                <MenuItem>
                  <Text>Item</Text>
                </MenuItem>
              </MenuToggle>
            </td>
          </tr>
          <tr>
            <td>Title 3</td>
            <td>Lorem ipsum dolor sit amet.</td>
            <td>
              <MenuToggle closeOnClick accessibilityLabel="Menu" toggleLabel="Actions">
                <MenuItem>
                  <Text>Item</Text>
                </MenuItem>
              </MenuToggle>
            </td>
          </tr>
        </tbody>
      </Table>
    </Text>
  );
}

withoutResponsiveWrapper.story = {
  name: 'Display a table without responsive wrapper (noWrap).',
};
