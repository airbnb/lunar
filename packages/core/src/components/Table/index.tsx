import React from 'react';
import withStyles, { WithStylesProps } from '../../composers/withStyles';
import withBoundary from '../../composers/withBoundary';
import Cell from './Cell';
import Row from './Row';

export type Props = {
  /** Apply a wrapping border. */
  bordered?: boolean;
  /** Table head, body, foot, and rows. */
  children: NonNullable<React.ReactNode>;
  /** Reduce padding within cells. */
  compact?: boolean;
  /** Disable automatic cell scaling. */
  fixed?: boolean;
  /** Apply a horizontal border between rows. */
  horizontal?: boolean;
  /** Mark as loading and disable interaction. */
  loading?: boolean;
  /** Applies verticalAlign: middle to child cells. */
  middleAlign?: boolean;
  /** Alternate row background color. */
  striped?: boolean;
  /** A unique name for tracking purposes. */
  trackingName?: string;
  /** Render with a transparent background. */
  transparent?: boolean;
  /** Apply a vertical border between rows. */
  vertical?: boolean;
};

/** A responsive table for displaying tabular data. */
export class Table extends React.Component<Props & WithStylesProps> {
  static defaultProps = {
    bordered: false,
    compact: false,
    fixed: false,
    horizontal: false,
    loading: false,
    middleAlign: false,
    striped: false,
    transparent: false,
    vertical: false,
  };

  render() {
    const {
      cx,
      bordered,
      children,
      compact,
      fixed,
      horizontal,
      loading,
      middleAlign,
      striped,
      styles,
      transparent,
      vertical,
    } = this.props;

    return (
      <div className={cx(styles.responsive_wrapper)}>
        <table
          className={cx(
            styles.table,
            middleAlign && styles.content_middle_align,
            fixed && styles.table_fixed,
            bordered && styles.table_bordered,
            horizontal && styles.table_horizontal,
            vertical && styles.table_vertical,
            compact && styles.table_compact,
            striped && styles.table_striped,
            loading && styles.table_loading,
            transparent && styles.table_transparent,
          )}
        >
          {children}
        </table>
      </div>
    );
  }
}

export { Cell, Row };

export default withBoundary('Table')(
  withStyles(({ color, ui, unit }) => {
    function createCell(styles: any) {
      return {
        '@selectors': {
          ':only-child td': styles,
          ':only-child th': styles,
        },
      };
    }

    return {
      table: {
        width: '100%',
        maxWidth: '100%',
        margin: 0,
        padding: 0,
        backgroundColor: color.accent.bg,
        border: '1px solid transparent',
        borderCollapse: 'collapse',
        borderSpacing: 0,

        '@selectors': {
          ':only-child td': {
            padding: unit * 1.5,
            verticalAlign: 'top',
          },

          ':only-child th': {
            padding: unit * 1.5,
            verticalAlign: 'bottom',
            whiteSpace: 'nowrap',
          },
        },
      },

      table_bordered: {
        border: ui.border,
      },

      table_compact: {
        ...createCell({
          padding: unit,
        }),
      },

      table_horizontal: {
        '@selectors': {
          ':only-child > tbody > tr > td': {
            borderTop: ui.border,
          },
        },
      },

      table_fixed: {
        display: 'table',
        tableLayout: 'fixed',
      },

      table_loading: {
        pointerEvents: 'none',
        opacity: 0.5,
      },

      table_striped: {
        '@selectors': {
          ':only-child > tbody > tr': {
            backgroundColor: color.accent.bg,

            '@selectors': {
              ':nth-child(odd)': {
                backgroundColor: color.accent.bgHover,
              },
            },
          },
        },
      },

      table_transparent: {
        backgroundColor: 'transparent',
      },

      table_vertical: {
        ...createCell({
          borderLeft: ui.border,
          borderRight: ui.border,
        }),
      },

      responsive_wrapper: {
        maxWidth: '100%',
        overflowX: 'auto',
      },
      content_middle_align: {
        '@selectors': {
          ':only-child td': {
            verticalAlign: 'middle',
          },
        },
      },
    };
  })(Table),
);
