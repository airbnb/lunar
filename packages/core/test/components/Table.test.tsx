import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { unwrapHOCs } from '@airbnb/lunar-test-utils';
import Table from '../../src/components/Table';

function unwrap(element: any): Enzyme.ShallowWrapper {
  // Dont use shallowWithStyles because of TrackingBoundary
  return unwrapHOCs(shallow(element), 'Table', {}, { render: true });
}

describe('<Table />', () => {
  it('renders bordered', () => {
    const wrapper = unwrap(<Table bordered>Bordered</Table>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders compact', () => {
    const wrapper = unwrap(<Table compact>Compact</Table>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders fixed', () => {
    const wrapper = unwrap(<Table fixed>Fixed</Table>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders horizontal', () => {
    const wrapper = unwrap(<Table horizontal>Horizontal</Table>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders loading', () => {
    const wrapper = unwrap(<Table loading>Loading</Table>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders striped', () => {
    const wrapper = unwrap(<Table striped>Striped</Table>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders transparent', () => {
    const wrapper = unwrap(<Table transparent>Transparent</Table>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders vertical', () => {
    const wrapper = unwrap(<Table vertical>Vertical</Table>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders middleAlign', () => {
    const wrapper = unwrap(<Table middleAlign>Vertical</Table>);

    expect(wrapper).toMatchSnapshot();
  });
});
