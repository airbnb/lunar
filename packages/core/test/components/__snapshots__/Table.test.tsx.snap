// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders bordered 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-table_bordered_1urjdog"
  >
    Bordered
  </table>
</div>
`;

exports[`<Table /> renders compact 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-table_compact_3oir43"
  >
    Compact
  </table>
</div>
`;

exports[`<Table /> renders fixed 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-table_fixed_91eznd"
  >
    Fixed
  </table>
</div>
`;

exports[`<Table /> renders horizontal 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-table_horizontal_1qi915e"
  >
    Horizontal
  </table>
</div>
`;

exports[`<Table /> renders loading 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-table_loading_e0u7o3"
  >
    Loading
  </table>
</div>
`;

exports[`<Table /> renders middleAlign 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-content_middle_align_14ilzse"
  >
    Vertical
  </table>
</div>
`;

exports[`<Table /> renders striped 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-table_striped_uvwezh"
  >
    Striped
  </table>
</div>
`;

exports[`<Table /> renders transparent 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-table_transparent_1eiqo0c"
  >
    Transparent
  </table>
</div>
`;

exports[`<Table /> renders vertical 1`] = `
<div
  className="responsive_wrapper_18n6f1e"
>
  <table
    className="table_1te9n79-o_O-table_vertical_dhjq9v"
  >
    Vertical
  </table>
</div>
`;
